airflow/models/renderedtifields.py

```diff
@@ -81,6 +83,31 @@ def get_templated_fields(cls, ti: TaskInstance, session: Session = None) -> Opti
         else:
             return None
 
+    @classmethod
+    @provide_session
+    def get_k8s_pod_yaml(cls, ti: TaskInstance, session: Session = None) -> Optional[str]:
+        """
+        Get templated field for a TaskInstance from the RenderedTaskInstanceFields
+        table.
+
+        :param ti: Task Instance
+        :param session: SqlAlchemy Session
+        :return: Rendered Templated TI field
+        """
+        result = (
+            session.query(cls.rendered_fields)
```

Comment:
This is going to only select the rendered_fields, not the `k8s_pod_yaml` field.
```suggestion
            session.query(cls.k8s_pod_yaml)
```

You probably need to improve on the tests to cover this as this wasn't caught.